eng/packages/http-client-csharp-mgmt/emitter/src/resource-detection.ts

```diff
@@ -805,6 +805,11 @@ function applyArmProviderSchemaDecorator(
   codeModel: CodeModel,
   schema: ArmProviderSchema
 ): void {
+  if (!codeModel.clients || codeModel.clients.length === 0) {
+    throw new Error(
+      "ARM provider schema decorator cannot be applied because no clients are defined in the code model."
+    );
+  }
```

Comment:
we should not error when there is no client, this is technically allowed.
Please remove this.